### PR TITLE
fix: surface provider errors in chat instead of generic connection failures

### PR DESCRIPTION
**Pull request by Wallgau**
issue: https://github.ibm.com/lakehouse/tracker/issues/77752
fix for #77752

Screenshot 2026-07-21 at 10 14 31 AM
Summary
- Surface real provider errors in chat via a structured stream error chunk
  (instead of “couldn’t connect to the chat service”)
- Format provider-health banner errors (e.g. IBM IAM) into a short message
  instead of dumping raw JSON; dedupe identical LLM + embedding failures
- Don’t create fake `error_*` conversation IDs on failed streams; only
  persist onto a real response id or a thread already loaded in memory

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved chat and onboarding streaming error flows with provider-safe formatting, early error short-circuiting, and correct terminal failure chunks.
  * Prevented duplicate assistant error messages and avoided clobbering locally appended chat turns after failed sends.
  * Refined provider health banner labeling when multiple validations fail.
  * Enhanced task failure toast descriptions with clearer, credential-aware messaging.
* **Tests**
  * Added/expanded unit tests for provider error detection/sanitization and streaming failure edge cases.
  * Updated Playwright checks to use more reliable error-row selectors.
* **Chores**
  * Fixed dev server hostname binding in the build script.

### CODE REVIEW

**Review by lucaseduoli — CHANGES_REQUESTED** [review]
image

When ingesting with a revoked API key, this shows instead of a message saying that the API key does not work

Captura de Tela 2026-07-21 às 14 17 42

The error message is leaking JSON on the chat. And the error design is just showing when sending the message, it's not shown if I refresh the page, it appears as a common message


And when I'm on an existing chat and send a message (with the revoked api), the message just disappears, and when I come back, it's duplicated

**Comment by edwinjosechittilappilly** [comment]
@Wallgau  what about the ingestion failures and failures at the time of connection.
@coderabbitai can you check if these cases are covered ?

File: frontend/lib/chat-stream-errors.ts [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    trimmed.includes("{") &&
+    (lowered.includes('"error"') ||
+      lowered.includes('"errormessage"') ||
+      lowered.includes('"errorcode"') ||
+      lowered.includes('"errors"'))
+  ) {
+    return true;
+  }
+  return false;
+}
```
[the remark this answers is not in the document]
Comment (edwinjosechittilappilly) [reply frontend/lib/chat-stream-errors.ts:166]: @Wallgau

[the remark this answers is not in the document]
Comment (Wallgau) [reply frontend/lib/chat-stream-errors.ts:166]: @coderabbitai check it is fixed now -> Exact match on opaque phrases like An error occurred while generating a response. is intentional and matches backend looks_like_provider_error_content → is_generic_upstream_error.

**Comment by Wallgau** [comment]
https://github.ibm.com/lakehouse/tracker/issues/77752

**Review by lucaseduoli — APPROVED** [review]
LGTM!
